app/src/main/res/layout/item_search_result.xml

```diff
@@ -48,9 +48,26 @@
             android:ellipsize="end"
             android:maxLines="1"
             app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/delete"
             app:layout_constraintStart_toEndOf="@id/searchResultIcon"
             app:layout_constraintTop_toTopOf="parent"
             tools:text="Search test" />
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/delete"
+            style="@style/IconButton"
+            android:layout_width="0dp"
+            android:layout_height="0dp"
+            android:layout_margin="@dimen/marginStandardSmall"
```

Comment:
Replace this margin with insets in all four directions so you can't click on the right side of the search item and miss the delete button